rules/otbn.bzl

```diff
@@ -185,6 +185,46 @@ def _otbn_sim_test(ctx):
     # The simulator and .elf file must be added to runfiles in order to be
     # visible to the test at runtime.
     runfiles = ctx.runfiles(files = (ctx.files.srcs + [elf, ctx.file._simulator]))
+    return [
+        DefaultInfo(runfiles = runfiles, executable = script_file),
+        providers[1],
+    ]
+
+def _otbn_consttime_test(ctx):
+    """This rule checks if a program or subroutine is constant-time.
+
+    There are some limitations to this check; see the Python script's
+    documentation for details. In particular, the check may not be able to
+    determine that a program runs in constant-time when in fact it does.
+    However, if the check passes, the program should always run in constant
+    time; that is, the check can produce false negatives but never false
+    positives.
+
+    This rule expects one dependency of an otbn_binary or otbn_sim_test type,
+    which should provide exactly one `.elf` file.
+    """
+
+    # Extract the output .elf file from the output group.
+    elf = [f for t in ctx.attr.deps for f in t[OutputGroupInfo].elf.to_list()]
+    if len(elf) != 1:
+        fail("Expected only one .elf file in dependencies, got: " + str(elf))
+    elf = elf[0]
+
+    # Write a very simple script that runs the checker.
+    script_file = ctx.actions.declare_file("{}.sh".format(ctx.attr.name))
+    script_content = "{} {} --verbose".format(ctx.executable._checker.short_path, elf.short_path)
+    if ctx.attr.subroutine:
+        script_content += " --subroutine {}".format(ctx.attr.subroutine)
+    if ctx.attr.secrets:
+        script_content += " --secrets {}".format(" ".join(ctx.attr.secrets))
+    ctx.actions.write(
+        output = script_file,
+        content = script_content,
+    )
+
+    # The checker and .elf file must be added to runfiles in order to be
+    # visible to the test at runtime.
+    runfiles = ctx.runfiles(files = (ctx.files._checker + [elf]))
```

Comment:
Try this:

```
runfiles = ctx.runfiles(files = ctx.files._checker + ctx.attr._checker[PyRuntimeInfo].files.to_list() + [elf])
```

What is happening is that `ctx.files._checker` refers to the files of the checker itself, but does not contain the files that comprise the Python runtime (e.g. the interpreter and all of it's dependencies).  The `PyRuntimeInfo` provider supplies the dependencies of the interpreter.

BTW, the way I found this was classic `print`-style debuging:
```
print(ctx.attr._checker) # Gives me info about the attribute, including the providers.
print(ctx.attr._checker[PyInfo]) # Prints out the provider itself, you can use `dir` if you don't recognize the provider.
print(dir(ctx.attr._checker[PyRuntimeInfo])) # Print out the attributes of the provider.
# ah, PyRuntimeInfo contains the python interpreter as a depset.
```

Reply:
That works, thank you! And thanks for the extra debug hints also, I'm still getting the hang of custom Bazel rules :slightly_smiling_face: 

I'll clean this up and move it out of draft shortly.